`superfluous-parens` false positive for `is not` tests

### Bug description

I just got the lint `C0325: Unnecessary parens after 'not' keyword (superfluous-parens)`, so I removed the parens it referred to, but thankfully a unit test caught that these are not the same, and the parens are needed (removing them changes the logic!).

I have been able to create a minimal example that I believe reproduces this

```python
In [1]: (None is None) is not (None == 1)  # Original, pylint lints with C0325
Out[1]: True

In [1]: (None is None) is not None == 1  # Recommended by C0325
Out[1]: False
```

### Configuration

_No response_

### Command used

```shell
pylint path/to/pkg/ --max-line-length=120 --fail-under=9 --load-plugins pylint_django
```


### Pylint output

```shell
************* Module pkg.module
path/to/pkg/module.py:63:0: C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
```


### Expected behavior

C0325 should not be output here.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6
Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -184,6 +184,10 @@ Release date: TBA
 
   Closes #4324
 
+* Avoid reporting ``superfluous-parens`` on expressions using the ``is not`` operator.
+
+  Closes #5930
+
 * Added the ``super-without-brackets`` checker, raised when a super call is missing its brackets.
 
   Closes #4008

--- a/doc/whatsnew/2.13.rst
+++ b/doc/whatsnew/2.13.rst
@@ -611,6 +611,10 @@ Other Changes
 
   Closes #5931
 
+* Avoid reporting ``superfluous-parens`` on expressions using the ``is not`` operator.
+
+  Closes #5930
+
 * Fix a crash when linting a file that passes an integer ``mode=`` to
   ``open``
 

--- a/pylint/checkers/format.py
+++ b/pylint/checkers/format.py
@@ -338,7 +338,11 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
             self._bracket_stack.pop()
         if tokens[start + 1].string != "(":
             return
-        if tokens[start].string == "not" and start > 0 and tokens[start - 1].string == "is":
+        if (
+            tokens[start].string == "not"
+            and start > 0
+            and tokens[start - 1].string == "is"
+        ):
             # If this is part of an `is not` expression, we have a binary operator
             # so the parentheses are not necessarily redundant.
             return

--- a/pylint/checkers/format.py
+++ b/pylint/checkers/format.py
@@ -338,6 +338,10 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
             self._bracket_stack.pop()
         if tokens[start + 1].string != "(":
             return
+        if tokens[start].string == "not" and start > 0 and tokens[start - 1].string == "is":
+            # If this is part of an `is not` expression, we have a binary operator
+            # so the parentheses are not necessarily redundant.
+            return
         found_and_or = False
         contains_walrus_operator = False
         walrus_operator_depth = 0
@@ -411,7 +415,7 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
                 elif token[1] == "for":
                     return
                 # A generator expression can have an 'else' token in it.
-                # We check the rest of the tokens to see if any problems incur after
+                # We check the rest of the tokens to see if any problems occur after
                 # the 'else'.
                 elif token[1] == "else":
                     if "(" in (i.string for i in tokens[i:]):

--- a/tests/functional/s/superfluous_parens.py
+++ b/tests/functional/s/superfluous_parens.py
@@ -68,3 +68,10 @@ class ClassA:
 
     def __iter__(self):
         return ((k, getattr(self, k)) for k in self.keys)
+
+if (A == 2) is not (B == 2):
+    pass
+
+M = A is not (A <= H)
+M = True is not (M == K)
+M = True is not (True is not False)

